From: tcmichals <tcmichals@yahoo.com>
Date: Sat, 22 Aug 2026 20:45:00 -0500
Subject: [PATCH 2/2] remoteproc: sunxi: add allwinner riscv remoteproc driver

Adds Linux remoteproc driver for the XuanTie E907/E906 co-processor on
Allwinner A523/A527/T527 and A733 SoCs. Manages Main CCU CLK_DSP (0x02001c70),
MCU CCU clock gates/resets, and loads firmware to dedicated RISC-V SRAM
(0x07280000, 256 KB) and System Shared SRAM C (0x00020000, 128 KB).

Signed-off-by: tcmichals <tcmichals@yahoo.com>
---
 drivers/remoteproc/Kconfig       |  10 +++
 drivers/remoteproc/Makefile      |   1 +
 drivers/remoteproc/sunxi_rproc.c | 175 +++++++++++++++++++++++++++++++
 3 files changed, 186 insertions(+)
 create mode 100644 drivers/remoteproc/sunxi_rproc.c

diff --git a/drivers/remoteproc/Kconfig b/drivers/remoteproc/Kconfig
index 8e45f9a..2a3b4c5 100644
--- a/drivers/remoteproc/Kconfig
+++ b/drivers/remoteproc/Kconfig
@@ -322,4 +322,14 @@ config ST_REMOTEPROC
 	  Say y to enable support for ST's companion processor via the
 	  remote processor framework.
 
+config SUNXI_REMOTEPROC
+	tristate "Allwinner XuanTie RISC-V remoteproc support"
+	depends objections (ARCH_SUNXI || COMPILE_TEST)
+	default y
+	help
+	  Say y or m here to support the Allwinner XuanTie E907/E906 RISC-V
+	  co-processor on Allwinner A523, A527, T527, and A733 SoCs via the
+	  remote processor framework.
+
 endif # REMOTEPROC
diff --git a/drivers/remoteproc/Makefile b/drivers/remoteproc/Makefile
index 7a8b9c0..3d4e5f6 100644
--- a/drivers/remoteproc/Makefile
+++ b/drivers/remoteproc/Makefile
@@ -34,3 +34,4 @@ obj-$(CONFIG_ST_REMOTEPROC)		+= st_remoteproc.o
 obj-$(CONFIG_STM32_RPROC)		+= stm32_rproc.o
 obj-$(CONFIG_TI_K3_DSP_REMOTEPROC)	+= ti_k3_dsp_remoteproc.o
 obj-$(CONFIG_TI_K3_R5_REMOTEPROC)	+= ti_k3_r5_remoteproc.o
+obj-$(CONFIG_SUNXI_REMOTEPROC)		+= sunxi_rproc.o
-- 
2.43.0
